tools/utils/learn_bpe.lua

```diff
@@ -0,0 +1,247 @@
+require('torch')
+
+local unicode = require 'tools.utils.unicode'
+local tokenizer = require('tools.utils.tokenizer')
+local case = require ('tools.utils.case')
+local separators = require('tools.utils.separators')
+
+local cmd = torch.CmdLine()
+
+cmd:text("")
+cmd:text("**learn_bpe.lua**")
+cmd:text("")
+
+cmd:option('-input', '', [[Input file for bpe learning]])
```

Comment:
`-input` is not consistent with other module - I would use STDIN for input but on the other hand add a `-save_bpe` for saving bpe model